src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -0,0 +1,233 @@
+            string targetFrameworkVersion = await commonProperties.GetEvaluatedPropertyValueAsync(ConfigurationGeneral.TargetFrameworkVersionProperty);
+
+            if (targetFrameworkVersion.StartsWith("v", StringComparison.OrdinalIgnoreCase))
+            {
+                targetFrameworkVersion = targetFrameworkVersion.Substring(startIndex: 1);
+            }
+
+            return targetFrameworkVersion;
+        }
+
+        private async Task ProcessExitedAsync(HotReloadState hotReloadState)
+        {
+            Assumes.NotNull(hotReloadState.Session);
+            Assumes.NotNull(hotReloadState.Process);
+
+            await _hotReloadDiagnosticOutputService.Value.WriteLineAsync($"{hotReloadState.Session.Name}: The process has exited.");
+
+            await StopProjectAsync(hotReloadState, default);
+        }
+
+        private IDeltaApplier? GetDeltaApplier(HotReloadState hotReloadState)
+        {
+            return null;
+        }
+
+        private Task<bool> RestartProjectAsync(HotReloadState hotReloadState, CancellationToken cancellationToken)
+        {
+            // TODO: Support restarting the project.
+            return TaskResult.False;
+        }
+
+        private Task OnAfterChangesAppliedAsync(HotReloadState hotReloadState, CancellationToken cancellationToken)
+        {
+            return Task.CompletedTask;
+        }
+
+        private async Task<bool> StopProjectAsync(HotReloadState hotReloadState, CancellationToken cancellationToken)
+        {
+            Assumes.NotNull(hotReloadState.Session);
+            Assumes.NotNull(hotReloadState.Process);
+
+            try
+            {
+                await hotReloadState.Session.StopSessionAsync(default);
+
+                ImmutableInterlocked.TryRemove(ref _activeSessions, hotReloadState.Process.Id, out _);
+            }
+            catch (Exception ex)
+            {
+                await _hotReloadDiagnosticOutputService.Value.WriteLineAsync($"{hotReloadState.Session.Name}: Error while stopping the session:\r\n{ex.GetType()}\r\n{ex.Message}");
+            }
+
+            return true;
+        }
+
+        private class HotReloadState : IProjectHotReloadSessionCallback
+        {
+            private readonly ProjectHotReloadSessionManager _sessionManager;
+
+            public Process? Process { get; set; }
+            public IProjectHotReloadSession? Session { get; set; }
+
+            // TODO: Support restarting the session.
+            public bool SupportsRestart => false;
+
+            public HotReloadState(ProjectHotReloadSessionManager sessionManager)
+            {
+                _sessionManager = sessionManager;
+            }
+
+#pragma warning disable VSTHRD100 // Avoid async void methods
+            internal async void OnProcessExited(object sender, EventArgs e)
+#pragma warning restore VSTHRD100 // Avoid async void methods
+            {
+                await _sessionManager.ProcessExitedAsync(this);
```

Comment:
Why not `JTF.RunAsync` here, to avoid `async void`?

Reply:
Yeah, that could work, too.

Reply:
I ended up using `IProjectFaultHandlerService.Forget` to run the async code but still capture exceptions.